src/components/Class/ClassForm.js

```diff
@@ -0,0 +1,720 @@
+          }
+          i = i + 1;
+        }
+        classFields.date = moment.utc(newDate).format("YYYY-MM-DD");
+      } else {
+        classFields.date = classFields.date;
+      }
+    } else {
+      classFields.date = classFields.date;
+    }
+
+    //taking hours and minues from the time
+    classFields.start_time = `${classFields.start_time.getHours()}:${classFields.start_time.getMinutes()}`;
+
+    classFields.end_time = `${classFields.end_time.getHours()}:${classFields.end_time.getMinutes()}`;
+
+    //combining time and date
+    const classStartTime = moment(
+      `${classFields.date} ${classFields.start_time}`
+    );
+    const classEndTime = moment(`${classFields.date} ${classFields.end_time}`);
+
+    if (classStartTime.valueOf() >= classEndTime.valueOf()) {
+      console.log("Class end time must be later than class start time.");
+    }
+
+    //deleting partner_id when it's length is 0
+    // if (classFields.partner_id.length === 0) delete classFields.partner_id;
+
+    //deleting date as we have combined with time and we don't want date separately
+    delete classFields.date;
+    // delete classFields[date];
+
+    //adding combined date and time to start_time and end_time
+    classFields.start_time = `${moment(classStartTime).format(
+      "YYYY-MM-DDTHH:mm:ss"
+    )}Z`;
+    classFields.end_time = `${moment(classEndTime).format(
+      "YYYY-MM-DDTHH:mm:ss"
+    )}Z`;
+
+    var fieldsToSend;
+
+    if (classFields.type === "doubt_class") {
+      fieldsToSend = {
+        course_id: classFields.course_id,
+        exercise_id: classFields.exercise_id,
+        category_id: classFields.category_id,
+        pathway_id: classFields.pathway_id,
+        title: classFields.title,
+        description: classFields.description,
+        start_time: classFields.start_time,
+        end_time: classFields.end_time,
+        lang: classFields.lang,
+        type: classFields.type,
+      };
+    } else if (classFields.type === "batch") {
+      fieldsToSend = {
+        title: classFields.title,
+        description: classFields.description,
+        start_time: classFields.start_time,
+        end_time: classFields.end_time,
+        partner_id: classFields.partner_id,
+        category_id: classFields.category_id,
+        pathway_id: classFields.pathway_id,
+        lang: classFields.lang,
+        frequency: classFields.frequency,
+        type: classFields.type,
+        on_days: classFields.on_days,
+      };
+    }
+
+    if (!isEditMode) {
+      if (classFields.max_enrolment != "No Limit") {
+        //add max_enrolment field only if it is not No Limit
+        fieldsToSend.max_enrolment = classFields.max_enrolment;
+      }
+      createClass(fieldsToSend);
+    } else {
+      if (classFields.max_enrolment != "No Limit") {
+        //add max_enrolment field only if it is not No Limit
+        fieldsToSend.max_enrolment = classFields.max_enrolment;
+      }
+      editClass(fieldsToSend);
+    }
```

Comment:
Use the `if/else` only for what's different between the cases:
```suggestion
    if (classFields.max_enrolment != "No Limit") {
      //add max_enrolment field only if it is not No Limit
      fieldsToSend.max_enrolment = classFields.max_enrolment;
    }
    (!isEditMode ? createClass : editClass)(fieldsToSend);
```